entity-framework/core/managing-schemas/scaffolding/templates.md

```diff
@@ -0,0 +1,218 @@
+
+***
+
+If you did everything correctly, the collection navigation properties should now use `ObservableCollection<T>`.
+
+```C#
+public virtual ICollection<Album> Albums { get; } = new ObservableCollection<Album>();
+```
+
+## Updating templates
+
+When you add the default templates to your project, it creates a copy of them based on that version of EF Core. As bugs are fixed and features are added in subsequent versions of EF Core, your templates may become out of date. You should review the changes made in the EF Core templates and merge them into your customized templates.
+
+One way to review the changes made to the EF Core templates is to use git to compare them between versions. The following command will clone the EF Core repository and generate a diff of these files between versions 7.0.0 and 8.0.0.
+
+git clone --no-checkout https://github.com/dotnet/efcore.git
+cd efcore
+git diff v7.0.0 v8.0.0 -- src/EFCore.Design/Scaffolding/Internal/CSharpDbContextGenerator.tt src/EFCore.Design/Scaffolding/Internal/CSharpEntityTypeGenerator.tt
+
+Another way to review the changes is to download the two versions of [Microsoft.EntityFrameworkCore.Templates](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Templates) from NuGet, extract their contents (you can change the file extensions to .zip), and compare those files.
+
+Before adding the default templates to a new project, remember to update to the latest EF Core template package.
+
+```dotnetcli
+dotnet new update
+```
+
+## Advanced usage
+
+### Ignoring the input model
+
+The `Model` and `EntityType` parameters represent one possible way of mapping to the database. You can choose to ignore or change parts of the model. For example, the navigation names we provide may not be ideal, and you can replace them with your own when scaffolding the code. Other things like constraint names and index filters are only used by Migrations and can safely be omitted from the model if you don't intend to use Migrations with the scaffolded code. Likewise, you may want to omit sequences or default constraints if they're not used by your app.
+
+When making advanced changes like this, just make sure the resulting model remains compatible with the database. Reviewing the SQL generated by `dbContext.Database.GenerateCreateScript()` is a good way to validate this.
+
+### Entity configuration classes
+
+For large models, the OnModelCreating method of the DbContext class can become unmanageably large. One way to address this is to use `IEntityTypeConfiguration<T>` classes. See [Creating and configuring a model](xref:core/modeling/index#grouping-configuration) for more information about these classes.
+
+To scaffold these classes, you can use a third template called `EntityTypeConfiguration.t4`. Like the `EntityType.t4` template, it gets used for each entity type in the model and uses the `EntityType` template parameter.
```

Comment:
Where can we get a sample of this template?

Reply:
An example would be great 🥳

Reply:
+1 We hope to provide it (and other template flavors) eventually. Probably as a community-driven project.
Here's the gist of what it would look like:

```tt
<#@ parameter name="EntityType" type="Microsoft.EntityFrameworkCore.Metadata.IEntityType" #>
<#@ parameter name="NamespaceHint" type="System.String" #>
using Microsoft.EntityFrameworkCore;

namespace <#= NamespaceHint #>;

public class <#= EntityType.Name #>Configuration : IEntityTypeConfiguration<<#= EntityType.Name #>>
{
    public void Configure(EntityTypeBuilder<<#= EntityType.Name #>> builder)
    {
<#
    // TODO: Move code from DbContext.t4 to here
#>
    }
}
```

Reply:
I will have a play with it. I already have a DbContext splitting option that can use this.